proxy/auth.go

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) OpenFaaS Project 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package proxy
+
+import (
+	"net/http"
+
+	"github.com/openfaas/faas-cli/config"
+)
+
+//SetAuth sets basic auth for the given gateway
+func SetAuth(req *http.Request, gateway string) {
```

Comment:
This function feels orphaned to me. I know why it's here because it gives a sense of removing duplication, but I think we could do this in a cleaner way. I saw a similar pattern for user-agent and didn't like it either.

I'm thinking of our own request factory which adds the things we know we need.

Doesn't have to be part of this PR. @stefanprodan  @rgee0 - thoughts?

Reply:
Would like your thoughts.